File: tests/04_link_1.go
package main

import . "glyph/tests/interp"

func t() {
    c := []uint64{
        0x0000004000000004,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0x00000000fffffffa,
        0xffffffffffffffff,
    }
    i := []uint16{
        CPU_encode_op_jal_i64(7,0),
        CPU_encode_op_break(0),
        CPU_encode_op_movq_i64(0,1),
        CPU_encode_op_or_i64(0,0,0),
        CPU_encode_op_jtl_i64(7,0),
    }
    CPU_test("link_1", c, i);
}

func main() {
    t()
}

File: tests/04_link_1.py
#!/usr/bin/env python3

from interp import *

def t():
    c = [
        0x0000004000000004,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0x00000000fffffffa,
        0xffffffffffffffff,
    ]
    i = [
        cpu_encode_op_jal_i64(7,0),
        cpu_encode_op_break(0),
        cpu_encode_op_movq_i64(0,1),
        cpu_encode_op_or_i64(0,0,0),
        cpu_encode_op_jtl_i64(7,0),
    ]
    cpu_test("link_1", c, i)

t()
